portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,28 +33,99 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> messages;
   private static final String REQUEST_PARAMETER_COMMENT_INPUT = "comment-input";
-  private static final String REDIRECT_COMMENTS = "/#comments";
-
-  @Override
-  public void init() {
-    messages = new ArrayList<>();
-  }
+  private static final String REQUEST_PARAMETER_PICTURE_LINK = "picture-link";
+  private static final String REQUEST_PARAMETER_NUMBER_COMMENTS = "number-comments";
+  private static final String REDIRECT_COMMENTS = "/comments.html";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<Comment> comments = new ArrayList<>();
+
+    int numberComments = 0;
+    int maxNumberComments = getPositiveInt(request);
```

Comment:
Does getPositiveInt return -1 in the failed case? Hence the protection on Lines 48-50?

Can the returned value be 0? Line 66 might fail forever then, right?

Reply:
Okay, saw the code below. 
-1 is error case, yes. 
0 cannot be returned.